src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1602,16 +1602,15 @@ abstract class RefChecks extends Transform {
                 try {
                   val treated = lits.mapConserve { lit =>
                     val stringVal = lit.asInstanceOf[Literal].value.stringValue
-                    treeCopy.Literal(lit, Constant(StringContext.processEscapes(stringVal)))
+                    val k = Constant(StringContext.processEscapes(stringVal))
+                    treeCopy.Literal(lit, k).setType(ConstantType(k))
```

Comment:
Thanks for the fix!

Reply:
Thanks for the learning opportunity and thanks @lrytz for the analysis and thanks @retronym for letting us help out. Github just popped up their new "insert a suggestion" feature and I clicked "got it", but actually I don't got it. It says committers can take the suggestion with a click. How dangerous is that? I would go for "insert pun", however.

Reply:
```suggestion
                    // To avoid the backlash of backslash, taken literally by Literal, escapes are processed strictly (scala/bug#11196)
                    treeCopy.Literal(lit, k).setType(ConstantType(k))
```

Reply:
"Suggestions cannot be applied while the pull request is closed." But really cool, I would have done. Also I will preserve "backlash of backslash" in my heart.